geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/Z3IdxStrategy.scala

```diff
@@ -54,18 +54,23 @@ class Z3IdxStrategy(val filter: QueryFilter) extends Strategy with LazyLogging w
     output(s"Temporal filters: ${filtersToString(temporalFilters)}")
 
     // standardize the two key query arguments:  polygon and date-range
-    val geomsToCover = tryReduceGeometryFilter(geomFilters).flatMap(decomposeToGeometry)
+    // TODO GEOMESA-1215 this can handle OR'd geoms, but the query splitter won't currently send them
+    val geometryToCover =
+      filter.singlePrimary.flatMap(extractSingleGeometry(_, sft.getGeomField)).getOrElse(WholeWorldPolygon)
 
-    val collectionToCover: Geometry = geomsToCover match {
-      case Nil => null
-      case seq: Seq[Geometry] => new GeometryCollection(geomsToCover.toArray, geomsToCover.head.getFactory)
-    }
-
-    // since we don't apply a temporal filter, we pass offsetDuring to
+    // since we don't apply a temporal filter, we pass handleExclusiveBounds to
     // make sure we exclude the non-inclusive endpoints of a during filter.
     // note that this isn't completely accurate, as we only index down to the second
-    val interval = extractInterval(temporalFilters, dtgField, exclusive = true)
-    val geometryToCover = netGeom(collectionToCover)
+    val interval = {
+      // TODO GEOMESA-1215 this can handle OR'd intervals, but the query splitter won't currently send them
+      val intervals = for { dtg <- dtgField; filter <- andOption(temporalFilters) } yield {
+        extractIntervals(filter, dtg)
+      }
+      // note: because our filters were and'ed, there will be at most one interval
+      intervals.flatMap(_.headOption).getOrElse {
+        throw new RuntimeException(s"Couldn't extract interval from filters '${filtersToString(temporalFilters)}'")
+      }
+    }
```

Comment:
Ooohh... is there more code sharing which could happen between this and the STIdx?  

Well, maybe that's moot...